module/zfs/dsl_dataset.c

```diff
@@ -2427,17 +2427,20 @@ get_receive_resume_stats_impl(dsl_dataset_t *ds)
 		zio_cksum_t cksum;
 		fletcher_4_native_varsize(compressed, compressed_size, &cksum);
 
-		str = kmem_alloc(compressed_size * 2 + 1, KM_SLEEP);
+		size_t alloc_size = compressed_size * 2 + 1;
+		str = kmem_alloc(alloc_size, KM_SLEEP);
 		for (int i = 0; i < compressed_size; i++) {
-			(void) sprintf(str + i * 2, "%02x", compressed[i]);
+			size_t index = i * 2;
```

Comment:
I'd like to avoid naming this `index` since it conflicts with `index(3)` from the user space `<strings.h>`.  How about `offset`.